workflow/scripts/get-transcript-info.R

```diff
@@ -39,69 +40,88 @@ while ( class(mart)[[1]] != "Mart" ) {
+#Check if three_prime_activated is activated or else if transcipts are cononical
+if (has_canonical && three_prime_activated) {
+  attributes <- c(attributes, "transcript_is_canonical", "chromosome_name",
+    "transcript_mane_select", "ensembl_transcript_id_version")
+}else if (has_canonical) {
+     attributes <- c(attributes, "transcript_is_canonical")
 }
-
 t2g <- biomaRt::getBM(
-            attributes = attributes,
-            mart = mart,
-            useCache = FALSE
-            )
-if (!has_canonical) {
-  t2g <- t2g %>% add_column(transcript_is_canonical = NA)
+attributes = attributes,
+mart = mart,
+useCache = FALSE
+)
+# Set columns as NA if three_prime_activated is set to false or if the transcipts are not canonical
+if (!has_canonical || !three_prime_activated) {
+  t2g <- t2g %>% add_column(chromosome_name = NA, transcript_mane_select = NA,
+      ensembl_transcript_id_version = NA)
+}else if (!has_canonical) {
+   t2g <- t2g %>% add_column(transcript_is_canonical = NA)
 }
-
 t2g <- t2g %>%
-        rename( target_id = ensembl_transcript_id,
-                ens_gene = ensembl_gene_id,
-                ext_gene = external_gene_name,
-                gene_desc = description,
-                canonical = transcript_is_canonical
-                ) %>%
-        mutate_at(
-          vars(gene_desc),
-          function(values) { str_trim(map(values, function (v) { str_split(v, r"{\[}")[[1]][1]})) } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
-        ) %>%
-        mutate_at(
-          vars(canonical),
-          function(values) {
-            as_vector(
-              map(
-                str_trim(values), 
-                function(v) {
-                  if (is.na(v)) {
-                    NA
-                  } else if (v == "1") { 
-                    TRUE 
-                  } else if (v == "0") {
-                    FALSE
-                  }
-                }
-              )
-            )
+  rename(
+    target_id = ensembl_transcript_id,
+    ens_gene = ensembl_gene_id,
+    ext_gene = external_gene_name,
+    gene_desc = description,
+    canonical = transcript_is_canonical,
+    chromosome_name = chromosome_name,
+    transcript_mane_select = transcript_mane_select,
+    ensembl_transcript_id_version = ensembl_transcript_id_version,
+  ) %>%
+  mutate_at(
+    vars(gene_desc),
+    function(values) {
+      str_trim(map(values, function(v) {
+        str_split(v, r"{\[}")[[1]][1]
+      }))
+    } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
+  ) %>%
+  mutate_at(
+    vars(canonical,),
+    function(values) {
+      as_vector(
+        map(
+          str_trim(values),
+          function(v) {
+            if (is.na(v)) {
+              NA
+            } else if (v == "1") {
+              TRUE
+            } else if (v == "0") {
+              FALSE
+            }
           }
         )
-
+      )
+    }
+  )
```

Comment:
Shouldn't such a treatment be applied to mane_select as well?

Reply:
We are not using `mane_select` column in any of the output tables as we use canonical. So I am not sure if we need to add this for mane_select.